src/pykeen/optimizers/rsgd.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+
+"""Riemannian SGD optimizer
+
+Code originally written by Ivana Balažević
+https://github.com/ibalazevic/multirelational-poincare/blob/master/rsgd.py
+and used under the MIT License.
+"""
+
+import torch
+from torch.optim.optimizer import Optimizer, required
+
+from ..utils import full_p_exp_map
+
+__all__ = [
+    'RiemannianSGD',
+]
+
+
+class RiemannianSGD(Optimizer):
```

Comment:
can this be subclassed from [`SGD`](https://pytorch.org/docs/stable/_modules/torch/optim/sgd.html)?

Reply:
If you look into the implementation of SGD, it's pretty monolithic. I don't see where we could effectively make use of parts that are already implemented. Plus, there's tons of magic in it that I don't understand